pages/desktop/user.py

```diff
@@ -119,8 +119,15 @@ def click_update_account(self):
         self.selenium.find_element(*self._update_account_locator).click()
         WebDriverWait(self.selenium, self.timeout).until(lambda s: self.update_message == "Profile Updated")
 
-    def change_hide_email_state(self):
-        self.selenium.find_element(*self._hide_email_checkbox).click()
+    def hide_email_address(self):
+        email_checkbox = self.selenium.find_element(*self._email_checkbox)
+        if not email_checkbox.is_selected():
+                self.selenium.find_element(*self._email_checkbox).click()
```

Comment:
After clicking on the element, you could then wait for the new state of the checkbox to be what we now expect it to be, again ensuring that the test does not proceed unless the page is in the state that we expect.